### DISCUSSION TITLE
[Docker] Problem with Recurring import!

### DISCUSSION THREAD

**Discussion by ghost** [pimcore/pimcore#16664] [Q&A / Support]
You certainly know the problem. It's been mentioned often enough. Unfortunately no answer has helped me so far. 

How do I get the recurring import to run in the Docker version? 

Until now I always had to do everything manually. So, click on start and then via docker container terminal: "bin/console datahub:data-importer:process-queue-parallel --processes=5".

But I would like to use the “Recurring” variant. ![grafik](https://github.com/pimcore/pimcore/assets/111425602/8c3cc00c-bc81-4e52-a5be-cab81d6c0867) 

Until now, Nothing is happening!

**Comment by dpfaffenbauer** [comment]
The docker image comes without cron installed. You have to therefore enable that and configure a cron job described like here: https://pimcore.com/docs/platform/Data_Importer/Installation#cron-execution

Alternatively, you could also run this command with supervisord: `php bin/console datahub:data-importer:execute-cron`

**Reply by ghost** [reply]
Okay, i will check this out. Thank You!

**Reply by dpfaffenbauer** [reply]
I would go the supervisor route and configure another route:

Add a new entry to this file: https://github.com/pimcore/demo/blob/11.x/.docker/supervisord.conf that looks like this:

```
[program:data-importer]
command=bash -c 'php /var/www/html/bin/console datahub:data-importer:execute-cron'
autostart=true
autorestart=true
stdout_logfile=/dev/fd/1
stdout_logfile_maxbytes=0
redirect_stderr=true
```

**Reply by ghost** [reply]
The only problem is that you would also have to make an entry for "bin/console datahub:data-importer:process-queue-parallel --processes=5" too. It would be good to work with some kind of process dependency. This is ootb in Supervisord only possible with a little creativity ^^

**Reply by ghost** [reply]
A somewhat tricky solution!
You set the importer to every minute, for example:
![grafik](https://github.com/pimcore/pimcore/assets/111425602/8ce21769-7752-4ff3-a332-20e749bd49b0)
Then you add new entrys like this: 


```
[program:data-importer]
command=bash -c 'sleep 70 && exec php bin/console datahub:data-importer:execute-cron'
autostart=true
autorestart=true
stdout_logfile=/dev/fd/1
stdout_logfile_maxbytes=0
redirect_stderr=true

[program:process-queue]
command=bash -c 'sleep 75 && exec php bin/console datahub:data-importer:process-queue-parallel --processes=5'
autostart=true
autorestart=true
stdout_logfile=/dev/fd/1
stdout_logfile_maxbytes=0
redirect_stderr=true
```

**Reply by ghost** [reply]
Old Topic, but similar problem. Installing cron.. I also have the Docker installation. Guess you have to
install Cron in the container? Or Would it be better to edit the image here? https://hub.docker.com/layers/pimcore/pimcore/php8.2-supervisord-latest/images/sha256-67c59e9585b1a20a9631f0ac4d05080d2911ea92b6857f62db77a3fa0c6c473c
